### PR TITLE
fix: Register copiers in the DI container using tags

### PR DESCRIPTION
**Pull request by hectoras**
**Associated Jira issue:** [ADF-1148](https://oat-sa.atlassian.net/browse/ADF-1148)

**Related PRs:**
- https://github.com/oat-sa/extension-tao-dac-simple/pull/195
- https://github.com/oat-sa/extension-tao-item/pull/590
- https://github.com/oat-sa/extension-tao-mediamanager/pull/414

### CODE REVIEW

**Review by wazelin — APPROVED** [review]

File: models/classes/resources/Service/ClassCopier.php [hunk 1]
```diff
@@ -73,6 +73,17 @@ public function withPermissionCopier(PermissionCopierInterface $permissionCopier
...
     }
 
+    /**
+     * This method is to be used with tagged_iterator() from service providers
+     * (but only the last copier from the iterable is effectively applied).
+     */
+    public function withPermissionCopiers(iterable $copiers): void
+    {
+        foreach($copiers as $copier) {
+            $this->withPermissionCopier($copier);
```
Comment (gabrielfs7) [inline models/classes/resources/Service/ClassCopier.php:83]: why apply only the last one?

File: models/classes/resources/Service/InstanceCopier.php [hunk 2]
```diff
@@ -58,6 +58,17 @@ public function withPermissionCopier(PermissionCopierInterface $permissionCopier
...
     }
 
+    /**
+     * This method is to be used with tagged_iterator() from service providers
+     * (but only the last copier from the iterable is effectively applied).
+     */
+    public function withPermissionCopiers(iterable $copiers): void
+    {
+        foreach($copiers as $copier) {
+            $this->withPermissionCopier($copier);
```
Comment (gabrielfs7) [inline models/classes/resources/Service/InstanceCopier.php:68]: why apply only the last one and why this method is created if `withPermissionCopier` already exists.

File: models/classes/resources/Service/ClassCopier.php (hunk 1 above)
Comment (hectoras) [reply models/classes/resources/Service/ClassCopier.php:83]: The former copier did not support more than a single permission copier, here I'm just providing a "glue" proxy method for that. We can change the `copy` method to call all class copiers instead (that may need additional manual & unit testing, but probably would be the proper solution).

File: models/classes/resources/Service/InstanceCopier.php (hunk 2 above)
Comment (hectoras) [reply models/classes/resources/Service/InstanceCopier.php:68]: https://github.com/oat-sa/tao-core/pull/3591#discussion_r946409241

**Review by gabrielfs7 — APPROVED** [review]

**Comment by hectoras** [comment]
Released as [v50.22.1](https://github.com/oat-sa/tao-core/releases/tag/v50.22.1)

**Pull request merged**